x-pack/plugins/lens/public/editor_frame_service/embeddable/embeddable.tsx

```diff
@@ -207,6 +209,9 @@ export class Embeddable extends AbstractEmbeddable<LensEmbeddableInput, LensEmbe
         embeddable: this,
       });
     }
+    if (isLensControlsEvent(event)) {
+      this.chartControls = event.data;
```

Comment:
A bit weird seeing events pipeline used for this. 
Maybe it is possible _to forward a ref_ to controls instead ?

Reply:
Curious, why do you think it is weird?

Reply:
My take on this, not sure how much subjective is this, so curious about other's opinions:

I wouldn't expect that to get access to an api, I have to use an event system. Yes it works and can't be implemented as is, but getting it via reference to a child component seems like more common pattern to me? 

I think using events for this is abusing them for things that there are not meant to handle:
* Events: notifying about changes or interactions in a system and passing meta information about the change 
* Not events: exposing imperative handlers to underlying APIs.

Reply:
i agree this is weird ..... i would expect `event.data` to be serializable ....

why do we need to do this, why doesn't chart handle this internally ?

Reply:
> ... getting it via reference to a child component seems like more common pattern to me?

It is effectively a function reference, like:

```jsx
<input ref={ref => {}} />
```

Functional references are better than named or object references.

> i agree this is weird ..... i would expect event.data to be serializable ....

I'm not sure why you find it weird, it is a common pattern. If `event.data` had to be serializable, then we would need to use something else, but I'm not aware of this requirement.

> why do we need to do this, why doesn't chart handle this internally ?

I'm not sure what you mean. We need to expose a way for embeddable a way to create a .png from a chart. The chart handles the part it needs to handle, the embeddable handles its part.